### PR TITLE
STAC Item retrieval

### PR DESCRIPTION
**Pull request by andher1802**
This PR adds the stac_item property to the assets Class.
For doing this we refactored the pystac_client authenticated against the UP42 account and removed support for Python 3.7.

Items:
* [x] Ran test & live-tests
* [x] Implemented (new) unit tests
* [ ] Removed credentials
* [ ] Removed argument pointing to staging
* [ ] Updated [documentation](sdk.up42.com)

For release:
* [ ] Bumped version
* [ ] Added changelog
* [ ] Updated announcement banner

### CODE REVIEW

**Review by seedlit — COMMENTED** [review]
nice!
Left some comments/questions :)

File: up42/asset.py [hunk 1]
```diff
@@ -74,6 +77,25 @@ def stac_info(self) -> Union[dict, None]:
             )
         return stac_results
 
+    @property
+    def stac_items(self) -> Optional[pystac.ItemCollection]:
+        url = f"{self.auth._endpoint()}/v2/assets/stac"
+        pystac_client_aux = pystac_auth_client(auth=self.auth).open(url=url)
+        stac_search_parameters = {
+            "max_items": 50,
```
[the remark this answers is not in the document]
Comment (andher1802) [reply up42/asset.py:85]: Actually in this case we would only expect to have one result as the asset-id is unique. set it to 50 as a default, but we can change it as well.

File: up42/stac_client.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from pystac_client import Client
+
+from up42.auth import Auth
+
+
+class pystac_auth_client(Client):
+    def __init__(
+        self, *args, id="id", description="description", auth: Auth = None, **kwargs
```
[the remark this answers is not in the document]
Comment (andher1802) [reply up42/stac_client.py:8]: In this case, a default value is required, due to some inheritance from the pystac_client and the implementation of the classmethods.

File: tests/test_storage.py [hunk 3]
```diff
@@ -36,17 +37,9 @@ def test_pystac_client_property(storage_mock):
...
-@mock.patch("pystac_client.Client.open")
-def test_pystac_client_property_invalid_auth_token(pystac_open_mock, storage_mock):
-    pystac_open_mock.side_effect = pystac_client.exceptions.APIError()
-    with pytest.raises(pystac_client.exceptions.APIError):
-        storage_mock.pystac_client  # pylint: disable=pointless-statement
-
-    pystac_open_mock.side_effect = [pystac_client.exceptions.APIError(), mock.DEFAULT]
-    up42_pystac_client = storage_mock.pystac_client
-    isinstance(up42_pystac_client, pystac_client.Client)
+    time.sleep(1)
```
[the remark this answers is not in the document]
Comment (andher1802) [reply tests/test_storage.py]: Done!

File: up42/asset.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (andher1802) [reply up42/asset.py]: Done!

File: up42/stac_client.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (seedlit) [reply up42/stac_client.py]: I see. Then maybe consider making it optional.
Pylance was complaining locally :) 

`from typing import Optional`
`self, *args, id="id", description="description", auth: Optional[Auth], **kwargs`

**Review by seedlit — APPROVED** [review]
left a small comment.
LGTM!

**Pull request merged**